lib/shoryuken/processor.rb

```diff
@@ -1,15 +1,79 @@
-require 'json'
-
 module Shoryuken
   class Processor
-    include Celluloid
     include Util
 
-    def initialize(manager)
-      @manager = manager
+    BATCH_LIMIT = 10
+
+    def initialize(fetcher, polling_strategy)
+      @fetcher = fetcher
+      @polling_strategy = polling_strategy
+      # what is the worst thing that could happen because it's not atomic?
+      # is it possible that it evaluates to true even if we never set it to true?
+      @done = false
+    end
+
+    def start
+      # does updating and evaluating @thread has to be wrapped in mutex?
+      # can it throw exception?
+      # what should we do if it dies?
+      @thread ||= spawn_thread(&method(:loop))
+    end
+
+    def terminate(wait=false)
+      @done = true
+      # does updating and evaluating @thread has to be wrapped in mutex?
+      return if !@thread
+      @thread.value if wait
+    end
+
+    def kill
+      @done = true
+      # does updating and evaluating @thread has to be wrapped in mutex?
+      return if !@thread
+      @thread.raise ::Shoryuken::Shutdown
+    end
```

Comment:
@mariokostelac that's where ^ I see some advantage of using concurrent-ruby/thread-pool, it abstracts some of this logic `shutdown`, `kill`, `wait_for_termination` etc. 

Also, replacing dead processors automatically. Here, we would need to re-spawn in case the thread dies.